src/librustc_data_structures/profiling.rs

```diff
@@ -585,3 +598,83 @@ fn get_resident() -> Option<usize> {
         }
     }
 }
+
+fn pass_name_to_string_id(
+    profiler: &Arc<SelfProfiler>,
+    string_cache: &mut FxHashMap<String, StringId>,
+    llvm_name: &str,
+) -> StringId {
+    *string_cache.entry(llvm_name.to_string()).or_insert_with(|| profiler.alloc_string(llvm_name))
+}
+
+fn ir_name_to_string_id(
+    profiler: &Arc<SelfProfiler>,
+    string_cache: &mut FxHashMap<String, StringId>,
+    llvm_name: &str,
+) -> StringId {
+    *string_cache.entry(llvm_name.to_string()).or_insert_with(|| {
+        let demangled_ir_name = rustc_demangle::demangle(llvm_name).to_string();
+        profiler.alloc_string(demangled_ir_name.as_str())
+    })
+}
+
+fn llvm_args_to_string_id(
+    profiler: &Arc<SelfProfiler>,
+    string_cache: &mut FxHashMap<String, StringId>,
+    pass_name: &str,
+    ir_name: &str,
+) -> EventId {
+    const SEPARATOR_BYTE: &str = "\x1E";
```

Comment:
We shouldn't copy the value of the separator byte into this file. It should be public so we can just do `measureme::event_id::SEPARATOR_BYTE` here.

Reply:
do not know why I was sure that I needed to do an update in measureme to get it public :)